src/rmq/rmqamqp/rmqamqp_consumer.cpp

```diff
@@ -154,7 +180,11 @@ bsl::optional<rmqamqpt::BasicMethod> Consumer::cancel()
 {
     bsl::optional<rmqamqpt::BasicMethod> method;
 
-    if (d_state == CONSUMING) {
+    if (d_state == NOT_CONSUMING) {
+        d_state = CANCELLED;
+        return method;
+    }
```

Comment:
Highlighting this change for the PR history (Shynar already discussed this internally). This fixes an issue where calling cancel on a consumer while the consumer isn't actually consuming, for example due to a connection issue, can put it in a stuck state.